networkx/algorithms/dag.py

```diff
@@ -1223,37 +1224,86 @@ def dag_to_branching(G):
 
 @not_implemented_for("undirected")
 @nx._dispatchable
-def compute_v_structures(G):
-    """Iterate through the graph to compute all v-structures.
+def compute_colliders(G):
+    """In a DAG, if you have three nodes A, B, and C, and there are edges from A to C
+    and from B to C, then C is a collider. This means that both A and B are "causing" C
+    in the graph, and conditioning on C would open up an association between A and B.
+
+    Parameters
+    ----------
+    G : graph
+        A networkx `DiGraph`.
 
-    V-structures are triples in the directed graph where
-    two parent nodes point to the same child and the two parent nodes
-    are not adjacent.
+    Returns
+    -------
+    collider : iterator of tuples
+        Each collider is a 3-tuple with the parent, collider, and other parent.
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If `G` is an undirected graph.
+
+    Examples
+    --------
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(1, 2), (0, 5), (3, 1), (2, 4), (3, 1), (4, 5), (1, 5)])
+    >>> sorted(nx.compute_colliders(G))
+    [(0, 5, 1), (0, 5, 4), (1, 5, 4)]
+
+    See Also
+    --------
+    compute_v_structures
+
+    Notes
+    -----
+    `Wikipedia: Collider in causal graphs <https://en.wikipedia.org/wiki/Collider_(statistics)>`_
+    """
+    for node in G.nodes:
+        for p1, p2 in combinations(G.predecessors(node), 2):
+            p1, p2 = sorted([p1, p2])  # sorting parents for consistent output
```

Comment:
We can't sort nodes.  They can be unsortable when two nodes can't be compared:  
`G=nx.Graph([(1, 'a')])`   or a more cool case where you use graphs as nodes in another graph.

Luckily `combinations` produces p1, p2 in a order (the order of `G.predecessors(node)`). So I think you can remove the line with the sort.  If you take the `sorted` out of the tests too, you can do something like this to avoid the sorting.
```python
    >>> G = nx.DiGraph()
    >>> G.add_edges_from([(1, 2), (0, 5), (3, 1), (2, 4), (3, 1), (4, 5), (1, 5)])
    >>> colliders = nx.compute_colliders(G)
    >>> all(u in G.pred[c] and v in G.pred[c] for u, c, v in colliders)
    True
```